csharp/src/Google.Protobuf.Test/FieldMaskTreeTest.cs

```diff
@@ -432,5 +432,44 @@ public void Merge(bool useDynamicMessage)
             Assert.IsNotNull(destination.Payload);
         }
 
+        [Test]
+        [TestCase(true, null)]
+        [TestCase(false, 24)]
+        public void MergeWrapperFields(bool replaceMessageFields, long? expectedInt64Value)
+        {
+            // Instantiate a destination with wrapper-based field types.
+            var destination = new TestWellKnownTypes()
+            {
+                StringField = "Hello",
+                Int32Field = 12,
+                Int64Field = 24,
+                BoolField = null,
+            };
+
+            // Set up a targeted update.
+            var source = new TestWellKnownTypes()
+            {
+                StringField = "Hi",
+                Int64Field = null
+            };
+
+            Merge(new FieldMaskTree().AddFieldPath("string_field").AddFieldPath("int64_field"),
+                source,
+                destination,
+                new FieldMask.MergeOptions()
+                {
+                    ReplaceMessageFields = replaceMessageFields
+                },
+                false);
+
+            // Make sure only the targeted fields were updated.
```

Comment:
I would expect this to look at Int32Field and BoolField... given that StringField *is* targeted, I wouldn't expect it to be asserted in this section. I suspect we effectively want two commented sections:

- Make sure only the targeted fields were updated (Int32Field and BoolField haven't changed)
- "Prove that ReplaceMessageFields works for wrapper types" - I'd expect the same test for both StringField and Int64Field here (see earlier comment about mixing null and non-null behavior in one test).

Reply:
Please see comment above and https://github.com/protocolbuffers/protobuf/pull/9602/commits/c4e7df18593b613ca08884fc5b955adae438ca3a